src/CycloneDX.Core/BomUtils.cs

```diff
@@ -454,5 +518,34 @@ public static void EnumerateAllToolChoices(Bom bom, Action<ToolChoices> callback
                     callback(vuln.Tools);
             });
         }
+
+        public static void EnumerateAllDependencies(Bom bom, Action<Dependency> callback)
+        {
+            var q = new Queue<Dependency>();
+
+            
+            q.EnqueueMany(bom.Dependencies);
+            
+
+            while (q.Count > 0)
+            {
+                var currentDependency = q.Dequeue();
+                if (currentDependency != null)
+                {
+                    callback(currentDependency);
+
+                    q.EnqueueMany(currentDependency.Dependencies);
+                }
+            }
+        }
+
+        public static void EnumerateAllDatasetChoices(Bom bom, Action<DatasetChoices> callback)
+        {
+            var x = bom.Components?.Select(c => c.ModelCard?.ModelParameters?.Datasets).Where(o => o != null) ?? new List<DatasetChoices>();
```

Comment:
Don't you need to use EnumerateAllComponents to cover all components and not just the toplevel ones?

Reply:
That's probably better